### PR TITLE
VIS: Accept xlabel and ylabel for scatter and hexbin plots

### LINKED ISSUE #37001: VIS: Allow xlabel and ylabel in DataFrame.plot.scatter plot and hexbin plot
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [x] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample
```python
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import matplotlib as mpl

ax = (pd.DataFrame(np.random.randn(100, 2), columns=['A','B'])
    .plot.scatter(
        x='A',
        y='B',
        xlabel='my x label',
        ylabel='my y label',
    )
)
```

#### Problem description

In #34223 (released in 1.1.0), the ability to set `xlabel` and `ylabel` in `DataFrame.plot()` was added. This works fine when calling `plot()` directly or `df.plot.line()`. When calling `df.plot.scatter()` with these args, you still get the index labels as the axis labels, instead of the strings you pass in.

These args are not documented to be incompatible or unavailable on this function, and no error is printed.

Actual Output

#### Expected Output

The plot's axis labels should be the strings provided, "my x label" and "my y label".

#### Output of ``pd.show_versions()``

<details>

```
INSTALLED VERSIONS
------------------
commit           : db08276bc116c438d3fdee492026f8223584c477
python           : 3.8.5.final.0
python-bits      : 64
OS               : Darwin
OS-release       : 19.6.0
Version          : Darwin Kernel Version 19.6.0: Sun Jul  5 00:43:10 PDT 2020; root:xnu-6153.141.1~9/RELEASE_X86_64
machine          : x86_64
processor        : i386
byteorder        : little
LC_ALL           : None
LANG             : en_GB.UTF-8
LOCALE           : en_GB.UTF-8

pandas           : 1.1.3
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 50.3.0
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 2.11.2
IPython          : 7.18.1
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : 3.3.2
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : 1.5.2
sqlalchemy       : None
tables           : None
tabulate         : 0.8.7
xarray           : None
xlrd             : None
xlwt             : None
numba            : None
```
</details>

### PR DESCRIPTION
**Pull request by treuherz**
- [x] closes #37001
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by treuherz** [comment]
I'm a bit puzzled by the type annotations on these params (from #34827)

https://github.com/pandas-dev/pandas/blob/9cb372376fc78aa66e2559de919592007b74cfaa/pandas/plotting/_matplotlib/core.py#L105-L106

They're `Optional[Label]` but `Label = Optional[Hashable]`. I'm all for making the Optional-ness of the parameters explicit but I'm not sure if the nesting has any side-effects, or why `Label` is `Optional` in the first place.

**Comment by charlesdong1991** [comment]
> I'm not sure if the nesting has any side-effects 

emm, any comments on the "potential" side-effect it might have? @simonjayhawkins  I added to make it more explicit, probably also okay if only having `Label`?

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -3449,6 +3449,26 @@ def test_xlabel_ylabel_dataframe_single_plot(
...
+        "xlabel, ylabel",
+        [
+            (None, None),
+            ("X Label", None),
+            (None, "Y Label"),
+            ("X Label", "Y Label"),
+        ],
+    )
+    @pytest.mark.parametrize("kind", ["scatter", "hexbin"])
+    def test_xlabel_ylabel_dataframe_plane_plot(self, kind, xlabel, ylabel):
```
Comment (charlesdong1991) [inline pandas/tests/plotting/test_frame.py:3462]: can you add a issue number below this line for future reference?

**Review by charlesdong1991 — APPROVED** [review]
very nice!

only one minor comment

**Comment by jreback** [comment]
lgtm. @charlesdong1991 merge when updated / green.

**Comment by simonjayhawkins** [comment]
> emm, any comments on the "potential" side-effect it might have? @simonjayhawkins I added to make it more explicit, probably also okay if only having `xlabel: Label = None`?

not an issue. adding Optional is fine. Where the problem occurs is where we need to distinguish between user passing None as a valid label, I don't think that's relevant here.

```
from pandas._typing import Label
from typing import Optional

a: Optional[Label]
reveal_type(a)  # Revealed type is 'Union[typing.Hashable, None]'
```

**Comment by treuherz** [comment]
Not sure why that check failed, it's way outside this PR

**Comment by charlesdong1991** [comment]
hmm, this seems to be related to https://github.com/pandas-dev/pandas/pull/37024 

will let you know once it's fixed so that you could rebase, sorry @treuherz

**Comment by simonjayhawkins** [comment]
> hmm, this seems to be related to #37024
> 
> will let you know once it's fixed so that you could rebase, sorry @treuherz

cc @arw2019 from changes in #37050

**Comment by charlesdong1991** [comment]
OK @simonjayhawkins shall i revert that PR to get CI green for now?

**Comment by charlesdong1991** [comment]
very nice! congrats on your first PR on pandas! @treuherz

**Pull request merged**